src/generator.cs

```diff
@@ -4371,9 +4370,8 @@ void GenerateProperty (Type type, PropertyInfo pi, List<string> instance_fields_
 				print ("}");
 			}
 			if (pi.CanWrite) {
-				PrintPlatformAttributes (pi);
-				PrintPlatformAttributes (pi.GetSetMethod ());
-				PrintPreserveAttribute (pi.GetSetMethod ());
+				PrintAttributes (pi, platform:true);
```

Comment:
I am not entirely sold on this being incorrect, what happens if the setter comes in a new update? IIRC we have had some breaking cases like this. 

Also It is super uncommon to have Availability attribute in the property itself and on the getters/setters